tests/test_core.py

```diff
@@ -120,6 +129,19 @@ def test_run_command_failure(runner, command):
         runner.run_command(command)
 
 
+@requires_ssh
+@pytest.mark.parametrize('content', ['echo $HOME', 'echo $(hostname -f)'])
```

Comment:
it is also not okay with `echo $(hostname -f)` (that test fails as well), but I don't think it should care about that: it should never look at the content of the file (which is the point of this PR).

My guess is that there's something wrong with the path, but funny enough the result of `cat {path}` contains the input data, so I'm not sure what this is about.

For full test isolation, it would probably be better to use a temporary file (run `mktemp -p {runner.log_dir}` or just `mktemp`) to create it and get the path).